src/cpp/src/whisper/models/statefull_decoder.cpp

```diff
@@ -16,7 +16,22 @@ WhisperStatefullDecoder::WhisperStatefullDecoder(const std::filesystem::path& mo
 
     utils::apply_slice_before_matmul_transformation(model);
 
-    auto compiled_model = core.compile_model(model, device, properties);
+    if (device.find("NPU") != std::string::npos) {
+        m_is_npu = true;
+    }
+
+    ov::CompiledModel compiled_model;
+    if (m_is_npu) {
```

Comment:
```suggestion
    ov::CompiledModel compiled_model;
    // npu device
    if (device.find("NPU") != std::string::npos) {
```

`m_is_npu` member seems redundant as used in ctor only

Reply:
And why `.find` needed can't we just compare `device == "NPU"`?

Reply:
Fixed